pkg/controller/job/job_controller.go

```diff
@@ -2122,3 +2132,14 @@ func managedByExternalController(jobObj *batch.Job) *string {
 	}
 	return nil
 }
+
+func newlyCreated(job *batch.Job) bool {
+	structIterator := reflect.ValueOf(job.Status)
+	for i := 0; i < structIterator.NumField(); i++ {
+		val := structIterator.Field(i).Interface()
+		if !reflect.DeepEqual(val, reflect.Zero(structIterator.Field(i).Type()).Interface()) {
+			return false
+		}
+	}
+	return true
```

Comment:
This is ok, but rather brute-force. I think it is enough to check is `status.startTime=nil && no conditions`. This should cover all cases, unless some users manipulate the status, but if they do, then technically they could remove the whole status as well.

Reply:
cc @soltysh @atiratree if you folks have any suggestions about the best way of solving this.

Reply:
I don't think the processing order of events is guaranteed, job controller puts the events in a queue and there are multiple workers to dequeue tasks, so the event processing is concurrent.

So alternatives can be:

* add more checks, if feasible, to determine it's really "newly created"
* only call `removeBackoffRecord()` without touch the `jm.expectations`, so that at least we fix the bug.

WDYT?

Reply:
That makes sense! Changed as suggested.

Reply:
+1 for having it inside the deleteJob handler